src/descriptor/mod.rs

```diff
@@ -848,7 +978,153 @@ impl<Pk: MiniscriptKey> fmt::Display for Descriptor<Pk> {
             Descriptor::Sh(ref sub) => write!(f, "sh({})", sub),
             Descriptor::Wsh(ref sub) => write!(f, "wsh({})", sub),
             Descriptor::ShWsh(ref sub) => write!(f, "sh(wsh({}))", sub),
+            Descriptor::ShSortedMulti(ref smv) => write!(f, "sh({})", smv),
+            Descriptor::WshSortedMulti(ref smv) => write!(f, "wsh({})", smv),
+            Descriptor::ShWshSortedMulti(ref smv) => write!(f, "sh(wsh({}))", smv),
+        }
+    }
+}
+
+/// Contents of a "sortedmulti" descriptor
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct SortedMultiVec<Pk: MiniscriptKey, Ctx: ScriptContext> {
+    /// signatures required
+    pub k: usize,
+    /// public keys inside sorted Multi
+    pub pks: Vec<Pk>,
+    /// The current ScriptContext for sortedmulti
+    pub(crate) phantom: PhantomData<Ctx>,
+}
+
+impl<Pk: MiniscriptKey, Ctx: ScriptContext> SortedMultiVec<Pk, Ctx> {
+    fn new(k: usize, pks: &[Pk]) -> Result<Self, Error> {
```

Comment:
Let's take a `Vec<Pk>` here rather than taking a slice that we immediately promote to a vector

Reply:
Done